### DISCUSSION TITLE
New Variable - $(SPECFILE)

### DISCUSSION THREAD

**Discussion by dylancliche** [LLNL/merlin#422] [Ideas]
It would be great to have a new Merlin variable `$(SPECFILE)` which gives the filename of the workflow (similar to `$(SPECROOT)` giving the absolute path to the workflow).

To my knowledge, there is nothing in Merlin which saves the original filename of the specification file. The closest things are `$(MERLIN_SPEC_ORIGINAL_TEMPLATE)`, `$(MERLIN_SPEC_EXECUTED_RUN)`, and `$(MERLIN_SPEC_ARCHIVED_COPY)`. However, these use the name given to the workflow in the `description:` block and this is most likely not the same as the filename, or there could be multiple workflows in the same `$(SPECROOT)` with the same name in the `description:` block but unique filenames in the directory.

**Comment by lucpeterson** [comment]
This seems like it should be fixed. The intent is that the ORIGINAL_TEMPLATE be the same file that was run. If we renamed that file based on the name field in the file, you’re right: that’s bad. 

My opinion is that this is a bug and the various provenance files should have the name of the original file (but with the suffix changed eg with .orig.yaml). IE we should change the file names, not add a new variable
